docs/roles/developer/contracts/shell.md

```diff
@@ -153,50 +174,94 @@ If the account was used to authorize an example app like Greeting or Counter ava
 
 In both examples of the access keys above, `FullAccess` keys and `FunctionCall` access keys, there is a `nonce` associated with each key which tracks how many times it has been used. This helps the NEAR runtime distinguish between transactions and avoid processing duplicate transactions. [See this page for more detail](https://nomicon.io/ChainSpec/Transactions.html#transaction-ordering-example-using-pool-iterator) about how NEAR uses `nonce` values.
 
+## Create a subaccount
 
-## Create a sub account
+When working with contracts, it's a good idea to create subaccounts. This makes it easier to iterate on contract development, particularly when working with Rust smart contracts. If a smart contract has fundamentally changed (where new fields are added to structs, for instance) and deployed to the same account, the state can run into issues. Without going too much into detail, having a subaccount is a good way to abate this type of problem. As we'll show later, it's simple to delete and recreate accounts if anything goes wrong.
 
-When working with contracts, you will create sub-accounts on your main account. These sub accounts are created using NEAR Shell
-
-Let's create a new account that is controlled your main account. But let's do this in the context of deploying a contract
+Create a subaccount from your main account with:
 
 ```bash
-near create_account name-of-contract-account._???_.testnet --master-account _???_.testnet --helper-url https://helper.testnet.near.org
+near create-account my-contract.YOU.testnet --masterAccount YOU.testnet
 ```
 
-## Deploy and test a contract
+## Deploy a contract
 
 **First build** your contract
 
-```sh
+For AssemblyScript smart contracts, the common pattern looks like this:
+
+```bash
 # the following command may very, check your package.json file
 yarn build:contract
 ```
 
-**Then deploy** the contract
+For Rust smart contracts, there will often be a `build.sh` script in the project, or there will be a command with `cargo build` including a few flags.
 
-- We assume you already created `name-of-contract._???_.testnet` in a previous step
+After the contract is build, **deploy** the contract.
 
-```text
-near deploy --wasm-file out/contract.wasm --account-id name-of-contract-account._???_.testnet
+(We assume you already created `name-of-contract.YOU.testnet` in a previous step.)
+
+```bash
+near deploy --wasmFile out/contract.wasm --accountId my-contract.YOU.testnet
 ```
 
-**Finally test** one of the contract methods
+## Deploy and initialize a contract
 
-```text
-near call name-of-contract-account._???_.testnet nameOfContractMethod --account-id _???_.testnet
+Many contracts (especially those written in Rust) will have an initialization method that is typically run immediately after deploying. This method is oftentimes called `new` and takes a couple arguments.
```

Comment:
minor grammar edit... 

will have an initialization method that ~~is~~ typically run***s*** immediately after deploying.

Reply:
thanks, adding this to PR with another internal suggestion